src/main/haskell/kore/src/Kore/Unification/UnificationSolution.hs

```diff
@@ -0,0 +1,136 @@
+{-|
+Module      : Kore.Unification.UnificationSolution
+Description : Datastructures and functionality for performing unification on
+              Pure patterns
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+Maintainer  : vladimir.ciobanu@runtimeverification.com
+Stability   : experimental
+Portability : portable
+-}
+module Kore.Unification.UnificationSolution where
+
+import Data.Functor.Foldable
+import Kore.AST.Common
+import Kore.AST.PureML
+import Kore.AST.MLPatterns
+import Kore.IndexedModule.MetadataTools
+import Kore.Step.StepperAttributes
+       ( StepperAttributes )
+import Kore.Step.PatternAttributes
+       ( FunctionalProof (..) )
+
+type UnificationSubstitution level variable
+    = [(variable level, PureMLPattern level variable)]
+
+-- |'UnificationSolution' describes the solution of an unification problem,
+-- consisting of the unified term and the set of constraints (equalities)
+-- obtained during unification.
+data UnificationSolution level variable = UnificationSolution
+    { unificationSolutionTerm        :: !(PureMLPattern level variable)
```

Comment:
Using prefixed field names is [deprecated in the style guide](https://github.com/kframework/kore/wiki/Coding-Standards#records).

Reply:
I agree, but this was a refactoring, so renaming these should not be mandatory (though it may be nice).

However, it is likely that this entire structure can be removed when Vlad finishes his PR.

Reply:
You were right, I was easily able to remove the type. Also renamed the file to `Data.hs` instead.